sklearn/model_selection/_search.py

```diff
@@ -372,6 +373,25 @@ def check(self):
 class BaseSearchCV(MetaEstimatorMixin, BaseEstimator, metaclass=ABCMeta):
     """Abstract base class for hyper parameter search with cross-validation."""
 
+    _parameter_constraints: dict = {
+        "estimator": [HasMethods(["fit"])],
```

Comment:
Looking at the implementation of `BaseSearchCV`, should we expect estimator to have all of those methods?
```suggestion
        "estimator": [
            HasMethods([
                "fit",
                "score",
                "predict",
                "predict_proba",
                "predict_log_proba",
                "decision_function",
                "score",
                "score_samples",
                "transform",
                "inverse_transform",
            ])
        ],
```

Reply:
I don't think so. Some should delegate to the underlying estimator. For instance `predict_proba` does not exist for the regressor but you can use the grid-search. I think that we should be a lenient and only request for `fit`. We are halfway to being an estimator in this case.